crates/api/src/v1/mod.rs

```diff
@@ -1,6 +1,30 @@
 //! Types representing v1 of the Warg REST API.
 
+pub mod content;
 pub mod fetch;
 pub mod package;
 pub mod paths;
 pub mod proof;
+
+use serde::{Deserialize, Serialize};
+
+/// Represents the supported kinds of content sources.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(tag = "type", rename_all = "camelCase")]
+pub enum ContentSource {
+    /// The content is located at an GET HTTP URL.
+    HttpGet {
+        /// The URL of the content.
+        url: String,
+        /// Optional support for HTTP Range header.
+        #[serde(default, skip_serializing_if = "is_false")]
+        supports_range_header: bool,
+        /// Optional content size in bytes.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        size: Option<u64>,
```

Comment:
Ok with me, more explicit.

Reply:
`size_hint` implies (to me) that it is an upper or lower bound, not an exact size.

Reply:
In practice, `size` should be the exact size. However, it isn't part of any verifiable data. I changed to `size_hint` but I can change back or rename it to something else. Let me know and I'll update the PR.

The intention of `supports_range_header` and `size` is giving the client enough info to determine if it should try to download in concurrent chunks instead of all at once.

Alternatively, the server could make the recommendation on whether to use the `Range` header and how many chunks (and what size). But I'm not sure I like that approach.

Reply:
I personally prefer, `size` over `size_hint`.

Reply:
I don't think it needs to be verifiable to be useful, but I do think we should specify whether or not it needs to be exact (if present).

Reply:
I pushed a change, renaming back to `size`.